File: core/src/main/java/com/cangwang/core/cwmodule/ex/CWAbsExModule.java
package com.cangwang.core.cwmodule.ex;

import android.os.Bundle;

import com.cangwang.core.MBaseApi;
import com.cangwang.core.cwmodule.CWModuleContext;
import com.cangwang.model.ICWModule;

/**
 * Created by cangwang on 2016/12/26.
 */

public abstract class CWAbsExModule implements ICWModule{

    public abstract boolean init(CWModuleContext moduleContext, Bundle extend);

    public abstract void onSaveInstanceState(Bundle outState);

    public abstract void onStart();

    public abstract void onResume();

    public abstract void onPause();

    public abstract void onStop();

    public abstract void onOrientationChanges(boolean isLandscape);

    public abstract void onDestroy();

    public abstract void detachView();

    public abstract void setVisible(boolean visible);

    public abstract void registerMApi(Class<?extends MBaseApi> key, MBaseApi value);

    public abstract void unregisterMApi(Class<?extends MBaseApi> key);

    public abstract boolean onBackPress();
}

File: core/src/main/java/com/cangwang/core/cwmodule/ex/CWAbsExModule.kt
package com.cangwang.core.cwmodule.ex

import android.os.Bundle
import com.cangwang.core.MBaseApi
import com.cangwang.core.cwmodule.CWModuleContext
import com.cangwang.model.ICWModule

/**
 * Created by cangwang on 2016/12/26.
 */
abstract class CWAbsExModule : ICWModule {
    abstract fun onCreate(moduleContext: CWModuleContext, extend: Bundle?): Boolean
    abstract fun onSaveInstanceState(outState: Bundle?)
    abstract fun onStart()
    abstract fun onResume()
    abstract fun onPause()
    abstract fun onStop()
    abstract fun onOrientationChanges(isLandscape: Boolean)
    abstract fun onDestroy()
    abstract fun detachView()
    abstract fun setVisible(visible: Boolean)
    abstract fun registerMApi(key: Class<out MBaseApi?>?, value: MBaseApi?)
    abstract fun unregisterMApi(key: Class<out MBaseApi?>?)
}